pyroomacoustics/room.py

```diff
@@ -2189,15 +2154,64 @@ def extrude(self, height):
 
         Room.extrude(self, np.array([0.0, 0.0, height]))
 
-        # update the shoebox dim
-        self.shoebox_dim = np.append(self.shoebox_dim, height)
-
     def get_volume(self):
 
-        """
-        Computes the volume of a room
-        :param room: the room object
-        :return: the volume in cubic unit
-        """
-
         return np.prod(self.shoebox_dim)
+
+
+class AnechoicRoom(ShoeBox):
+    """ 
+    This class provides an API for creating an Anechoic "room" in 2D or 3D.
+
+    Parameters
+    ----------
+    dim: int
+        Dimension of the room (2 or 3).
+    kwargs:
+        All other kwargs are passed on to the :py:obj:`pyroomacoustics.room.Room` initializer.
+    """
+
+    def __init__(self, dim, **kwargs):
+        forbidden = ["max_order", "ray_tracing", "walls", "p"]
+        for kw in forbidden:
+            if kw in kwargs.keys():
+                warnings.warn(
+                    "Ignoring for Anechoic room: {}={}".format(kw, kwargs[kw])
+                )
+
+        kwargs["max_order"] = 0
+        kwargs["ray_tracing"] = False
+
+        # Create some dummy walls
+        p = np.ones((dim,))
+        materials = Material(energy_absorption=1.0, scattering=0.0)  #
+
+        ShoeBox.__init__(self, p=p, materials=materials, **kwargs)
+
+    def __str__(self):
+        return "AnechoicRoom instance in {}D.".format(self.dim)
+
+    def is_inside(self, p):
+        """ Overloaded function to eliminate testing if objects are "inside" room. """
+        # always return True because we want the walls to have no effect.
+        return True
+
+    def get_bbox(self):
```

Comment:
I guess it depends on how you wrote the plotting function too, but should the bbox be infinite or made so that it contains all the microphones and sources ?

Reply:
That's how it is implemented for now: bbox is infinite by default, and then gets reduced to cover all microphones and sources

Reply:
I added an error because if there are neither mics nor sources, the bbox stays infinite and one cannot set infinite bounds in matplotlib.